src/metabase/async/util.clj

```diff
@@ -39,9 +39,7 @@
   ;; * `result-chan` will get the result of `(f)`, *after* `done-chan` is closed
   (let [done-chan   (a/promise-chan)
         result-chan (a/promise-chan)
-        binds       (clojure.lang.Var/getThreadBindingFrame)
-        thunk*      (^:once fn* []
-                      (clojure.lang.Var/resetThreadBindingFrame binds)
+        thunk*      (^:once fn []
```

Comment:
I don't quite get why this explicit bindings capture and setting exists here because a few lines below `thunk*` is still wrapped in `bound-fn*`. Digging into git history it looks like this is evolutionary; these bindings were not cleaned up after `bound-fn*` was added. Anyway, not so sure about this one.

Reply:
This one is over my head.

Reply:
Same here. I think this was accidental. Given that the function is currently only used in one place, I think it's worth simplifying it now while the potential impact is low.